packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/chat.tests.js

```diff
@@ -3148,5 +3148,51 @@ QUnit.module('Chat', () => {
             $indicator = this.$element.find(`.${SCROLLVIEW_REACHBOTTOM_INDICATOR}`);
             assert.strictEqual($indicator.is(':visible'), false, 'loading indicator is hidden');
         });
+
+        QUnit.test('should update only the necessary changes if the new Message object is passed to store.push', function(assert) {
+            const initialMessageData = { text: 'message_1', isDeleted: false };
+            const messages = [
+                { id: 1, ...initialMessageData },
+                { id: 2, text: 'message_2' }
+            ];
+            const timeout = 400;
+
+            const store = new CustomStore({
+                key: 'id',
+                load: function() {
+                    const d = $.Deferred();
+                    setTimeout(function() {
+                        d.resolve(messages);
+                    }, timeout);
+                    return d.promise();
+                },
+            });
+
+            this.reinit({ dataSource: store });
+
+            this.clock.tick(timeout);
+
+            const bubble = MessageBubble.getInstance(this.getBubbles().eq(0));
+            const updateContentSpy = sinon.spy(bubble, '_updateContent');
+
+            [
+                { optionsChanged: 0 },
+                { ...initialMessageData, optionsChanged: 0 },
+                { text: 'new text 1', isDeleted: false, optionsChanged: 1 },
+                { text: 'new text 1', isDeleted: true, optionsChanged: 1 },
+                { text: 'new text 2', isDeleted: true, optionsChanged: 1 },
+                { text: 'new text 3', isDeleted: false, optionsChanged: 2 },
+            ].forEach(dataToChange => {
```

Comment:
Lets split configuration and expected result as following
{ config: { text, isDeleted }, expected: { optionChangedCallCount: 1 } }